blocks/library/button/index.js

```diff
@@ -200,12 +204,33 @@ registerBlockType( 'core/button', {
 	save( { attributes } ) {
 		const { url, text, title, align, color, textColor } = attributes;
 
+		const buttonStyle = {
+			backgroundColor: color,
+			color: textColor,
+		};
+
+		const linkClass = 'wp-block-button__link';
+
 		return (
-			<div className={ `align${ align }` } style={ { backgroundColor: color } }>
-				<a href={ url } title={ title } style={ { color: textColor } }>
+			<div className={ `align${ align }` }>
+				<a className={ linkClass } href={ url } title={ title } style={ buttonStyle }>
 					{ text }
 				</a>
 			</div>
 		);
 	},
+
+	deprecated: [ {
```

Comment:
It's not working for me, I believe you should assign the attributes definition to the deprecated version as well.

Reply:
I tried this in cdcafc39829950084045f1f2e87ea34410a427f5 but I think it's still not working.

Reply:
It worked for me.

Reply:
Same here when tested again but sometimes it didn't for some reason:)

Reply:
I think there's probably a bug related to the deprecated blocks handling and the hooks (custom css class for example). But it's a separate issue.